### PR TITLE
[WIP] Abstract application of rules

### PR DESCRIPTION
**Pull request by ivansg44**
This addresses #9.

Similar code is used in rules 2-5 to fill several output fields:
- "matched_term"
- "all_matched_terms_with_resource_ids"
- "retained_terms_with_resource_ids"
- "number_of_components_for_component_match"
- "match_status_macro_level"
- "match_status_micro_level"

This code should be abstracted into a single function.

### CODE REVIEW

**Comment by ivansg44** [comment]
Build 76 has failed. This may be a Travis CI problem, as heir Github service is currently "partially degraded". Build logs and the fact local tests are passing support this.

We will keep an eye on this.

File: lexmapr/pipeline.py [hunk 1]
```diff
@@ -89,6 +89,7 @@ def find_left_r(s, first, last):
 
 
 # 7-Methods to add suffixes such as food product or product to input phrase to improve Term matching
+# TODO: Does a simple operation like this warrant a function?
```
Comment (dfornika) [inline lexmapr/pipeline.py]: It seems simpler to just do this in-line with `+ " " +` rather than define a function

Comment (ivansg44) [reply lexmapr/pipeline.py]: See 505f6eb.

**Comment by ivansg44** [comment]
@dfornika Do you think this PR is ready for merging? Do you see any other small, immediate changes I can make to better abstract rules application?

There is still some work left to do for find_full_term_match, as detailed in the TODO list of its function docstring. But this work will involve may involve larger changes to how data is stored in pipeline.py, and to the output test files. I think this would make the current PR less focused, so such changes may be better suited for a future PR. Thoughts?

**Comment by dfornika** [comment]
Looks ready to me. The only thing I can see is to remove that commented-out `statusAddendum` variable if it isn't being used.

**Comment by dfornika** [comment]
Shall we merge?

**Comment by Public-Health-Bioinformatics** [comment]
make it so!

**Comment by ivansg44** [comment]
We could return an empty ret when a match is not found instead. I just thought using an error would be more meaningful and intuitive than an empty ret value, and that we could use this error for future functions as well (e.g., for component-matching).

We can get rid of it if you want--otherwise, it's ready for merging.

**Pull request merged**